backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,246 @@
+            if (robot == null)
+            {
+                string errorMessage = $"Robot with ID: {robotId} was not found in the database";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            if (await missionRunService.OngoingMission(robot.Id)) { await WaitForLocalizationMissionStatusToBeUpdated(robot); }
+
+            var lastExecutedMissionRun = await missionRunService.ReadLastExecutedMissionRunByRobot(robot.Id);
+            if (lastExecutedMissionRun is null)
+            {
+                string errorMessage = $"Could not find last executed mission run for robot with ID {robot.Id}";
+                logger.LogError("{Message}", errorMessage);
+                throw new MissionNotFoundException(errorMessage);
+            }
+
+            if (lastExecutedMissionRun.Status != MissionStatus.Successful)
+            {
+                string errorMessage =
+                    $"The localization mission {lastExecutedMissionRun.Id} failed and thus subsequent scheduled missions for deck {lastExecutedMissionRun.Area?.Deck} wil be cancelled";
+                logger.LogError("{Message}", errorMessage);
+                throw new LocalizationFailedException(errorMessage);
+            }
+
+            await robotService.UpdateCurrentArea(robot.Id, lastExecutedMissionRun.Area);
+        }
+
+        private async Task WaitForLocalizationMissionStatusToBeUpdated(Robot robot)
+        {
+            if (robot.CurrentMissionId is null)
+            {
+                string errorMessage = $"Could not find current mission for robot {robot.Id}";
+                logger.LogError("{Message}", errorMessage);
+                throw new MissionNotFoundException(errorMessage);
+            }
+
+            string ongoingMissionRunId = robot.CurrentMissionId;
+            var ongoingMissionRun = await missionRunService.ReadById(robot.CurrentMissionId);
+            if (ongoingMissionRun is null)
+            {
+                string errorMessage = $"Could not find ongoing mission with ID {robot.CurrentMissionId}";
+                logger.LogError("{Message}", errorMessage);
+                throw new MissionNotFoundException(errorMessage);
+            }
+
+            if (!ongoingMissionRun.IsLocalizationMission())
+            {
+                string errorMessage = $"The currently executing mission for robot {robot.CurrentMissionId} is not a localization mission";
+                logger.LogError("{Message}", errorMessage);
+                throw new OngoingMissionNotLocalizationException(errorMessage);
+            }
+
+            logger.LogWarning(
+                "The RobotAvailable event was triggered before the OnMissionUpdate event and we have to wait to see that the localization mission is set to successful");
+
+            const int Timeout = 5;
+            var timer = new Stopwatch();
```

Comment:
There are ways of awaiting event handlers which we should use instead in the future, but this is probably fine for now.